addons/web_editor/static/tests/link_test.js

```diff
@@ -0,0 +1,458 @@
+        document.execCommand("insertText", false, char);
+        document
+            .querySelector(selector)
+            .dispatchEvent(new window.KeyboardEvent("keydown", { key: char }));
+        document
+            .querySelector(selector)
+            .dispatchEvent(new window.KeyboardEvent("keyup", { key: char }));
+    }
+}
+
+let serverData;
+let wysiwyg;
+
+QUnit.module(
+    "web_editor",
+    {
+        before: function () {
+            serverData = {
+                models: {
+                    note: {
+                        fields: {
+                            body: {
+                                string: "Editor",
+                                type: "html",
+                            },
+                        },
+                        records: [
+                            {
+                                id: 1,
+                                display_name: "first record",
+                                body: "<p><br></p>",
+                            },
+                        ],
+                    },
+                },
+            };
+        },
+        beforeEach: async function () {
+            setupViewRegistries();
+            patchWithCleanup(Wysiwyg.prototype, {
+                init() {
+                    super.init(...arguments);
+                    wysiwyg = this;
+                },
+            });
+            await makeView({
+                type: "form",
+                serverData,
+                resModel: "note",
+                arch:
+                    "<form>" +
+                    '<field name="body" widget="html" style="height: 100px"/>' +
+                    "</form>",
+                resId: 1,
+            });
+        },
+    },
+    function () {
+        QUnit.module("HotKeys");
+
+        QUnit.test("should be able to create link with ctrl+k and ctrl+k", async function (assert) {
+            const { editor, editable } = onMount();
+            const node = editable.querySelector("p");
+            setSelection(node, 0);
+            await triggerHotkey("control+k");
+            inputText(".o_command_palette_search", "k");
+            await nextTick();
+            triggerEvent(node, "keydown", {
+                key: "Enter",
+            });
+            await nextTick();
+            inputText('input[id="o_link_dialog_url_input"]', "#");
+            editor.document.querySelector(".o_dialog footer button.btn-primary").click();
+            await nextTick();
+            assert.strictEqual(
+                editable.innerHTML,
+                `<p><a href="#" target="_blank">#</a><br></p>`
+            );
+        });
+
+        QUnit.test("should be able to create link with ctrl+k and k", async function (assert) {
+            const { editor, editable } = onMount();
+            const node = editable.querySelector("p");
+            setSelection(node, 0);
+            await triggerHotkey("control+k");
+            await triggerHotkey("control+k")
```

Comment:
Nice! I know it's kind my fault here :grimacing: , but add a few comments to make it clear why ctrl+k is hit twice, something like "Open command palette" and "Hot key for Create link".